tasktiger/redis_scripts.py

```diff
@@ -469,20 +480,39 @@ def execute_pipeline(self, pipeline, client=None):
         results = redis_scripts.execute_pipeline(p)
         """
 
+        client = client or self.redis
+
+        executing_pipeline = None
         try:
+
             # Prepare args
             stack = pipeline.command_stack
-            script_args = [len(stack)]
+            script_args = [int(self.can_replicate_commands), len(stack)]
             for args, options in stack:
                 script_args += [len(args)-1] + list(args)
 
-            # Make sure scripts exist
-            if pipeline.scripts:
-                pipeline.load_scripts()
-
             # Run the pipeline
-            raw_results = self._execute_pipeline(args=script_args,
-                                                 client=client)
+            if self.can_replicate_commands:  # Redis 3.2 or higher
+                # Make sure scripts exist
+                if pipeline.scripts:
+                    pipeline.load_scripts()
+
+                raw_results = self._execute_pipeline(args=script_args,
+                                                     client=client)
+            else:
+                executing_pipeline = client.pipeline()
+
+                # Always load scripts to avoid issues when Redis loads data
+                # from AOF file / when replicating.
+                for s in pipeline.scripts:
+                    executing_pipeline.script_load(s.script)
```

Comment:
We should monitor Redis CPU and network bandwidth after deploying this change to see if there is any noticeable impact.